src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibilityHelp.ts

```diff
@@ -47,25 +47,23 @@ export class TerminalAccessibleContentProvider extends Disposable implements IAc
 		_xterm: Pick<IXtermTerminal, 'getFont' | 'shellIntegration'> & { raw: Terminal },
 		@IInstantiationService _instantiationService: IInstantiationService,
 		@IKeybindingService private readonly _keybindingService: IKeybindingService,
-		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
 		@IContextKeyService private readonly _contextKeyService: IContextKeyService,
-		@ICommandService private readonly _commandService: ICommandService
+		@ICommandService private readonly _commandService: ICommandService,
+		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService
 	) {
 		super();
 		this._hasShellIntegration = _xterm.shellIntegration.status === ShellIntegrationStatus.VSCode;
 	}
 
 	private _descriptionForCommand(commandId: string, msg: string, noKbMsg: string): string {
-		const kb = this._keybindingService.lookupKeybindings(commandId);
-		switch (kb.length) {
-			case 0:
-				return format(noKbMsg, commandId);
-			case 1:
-				return format(msg, kb[0].getAriaLabel());
+		if (commandId === TerminalCommandId.RunRecentCommand) {
+			const kb = this._keybindingService.lookupKeybindings(commandId);
+			// Run recent command has multiple keybindings. lookupKeybinding just returns the first one regardless of the when context.
```

Comment:
note this has been investigated in a separate issue assigned to someone else

Reply:
I think you're after `softDispatch` which is what commands to skip shell uses

Reply:
so i'd have to create an event just to get the right keybinding? that seems odd to me.

Reply:
In order to evaluate what keybinding would occur _with a given context_ you need to soft dispatch it afaik.

Reply:
there's no event in this case, so i'm not following